tests/test_buffering_wrapper.py

```diff
@@ -0,0 +1,215 @@
+    done = (t == self.episode_length)
+    return t, t * 10, done, {}
+
+
+def _make_buffering_venv(error_on_premature_reset: bool,
+                         ) -> BufferingWrapper:
+  venv = DummyVecEnv([_CountingEnv] * 2)
+  venv = BufferingWrapper(venv, error_on_premature_reset)
+  venv.reset()
+  return venv
+
+
+def _assert_equal_scrambled_vectors(a: np.ndarray, b: np.ndarray) -> bool:
+  """Returns True if `a` and `b` are identical up to sorting."""
+  assert a.shape == b.shape
+  assert a.ndim == 1
+  np.testing.assert_allclose(np.sort(a), np.sort(b))
+
+
+def _join_transitions(trans_list: Sequence[rollout.Transitions],
+                      ) -> rollout.Transitions:
+
+  def concat(x): return np.concatenate(list(x))
+  obs = concat(t.obs for t in trans_list)
+  next_obs = concat(t.next_obs for t in trans_list)
+  rews = concat(t.rews for t in trans_list)
+  acts = concat(t.acts for t in trans_list)
+  dones = concat(t.dones for t in trans_list)
+  return rollout.Transitions(
+    obs=obs, next_obs=next_obs, rews=rews, acts=acts, dones=dones)
+
+
+@pytest.mark.parametrize("episode_lengths", [(1,), (6, 5, 1, 2), (2, 2)])
+@pytest.mark.parametrize("n_steps", [1, 2, 20, 21])
+@pytest.mark.parametrize("extra_pop_timesteps", [(), (1,), (4, 8)])
+def test_pop(episode_lengths: Sequence[int],
+             n_steps: int,
+             extra_pop_timesteps: Sequence[int]):
+  """Check pop_transitions() results for BufferWrapper.
+
+  To make things easier to test, we use _CountingEnv where the observation
+  is simply the episode timestep. The reward is 10x the timestep. Our action
+  is 2.1x the timestep. There is an confusing offset for the observation because
+  it has timestep 0 (due to reset()) and the other quantities don't, so here is
+  an example of environment outputs and associated actions:
+
+  ```
+  episode_length = 5
+  obs = [0, 1, 2, 3, 4, 5]  (len=6)
+  acts = [0, 2.1, 4.2, ..., 8.4]  (len=5)
+  rews = [10, ..., 50]  (len=5)
+  ```
+
+  Converted to `Transition`-format, this looks like:
+  ```
+  episode_length = 5
+  obs = [0, 1, 2, 3, 4, 5]  (len=5)
+  next_obs = [1, 2, 3, 4, 5]  (len=5)
+  acts = [0, 2.1, 4.2, ..., 8.4]  (len=5)
+  rews = [10, ..., 50]  (len=5)
+  ```
+
+  Args:
+    episode_lengths: The number of timesteps before episode end in each dummy
+      environment.
+    n_steps: Number of times to call `step()` on the dummy environment.
+    extra_pop_timesteps: By default, we only call `pop_*()` after `n_steps`
+      calls to `step()`. For every unique positive `x` in `extra_pop_timesteps`,
+      we also call `pop_*()` after the `x`th call to `step()`. All popped
+      samples are concatenated before validating results at the end of this
+      test case. All `x` in `extra_pop_timesteps` must be in range(1, n_steps).
+      (`x == 0` is not valid because there are no transitions to pop at timestep
+      0).
+  """
+  if not n_steps >= 1:
+    raise ValueError(n_steps)
```

Comment:
Missing coverage on this line. I should change into `assert` or use `# nocov?something`.